### ISSUE TITLE
Why is the workflow run so long ?

### ISSUE THREAD

**Issue by OrangeCatLoves** [nus-cs2103-AY2425S1/forum#527]
![image](https://github.com/user-attachments/assets/523d439d-55fb-45d0-abe4-63be8cc9ebad)
For some reason, the run for ubuntu-latest is taking forever. It might be because previously I pushed a change (added 2 new lines of dependencies to build.gradle file) and it caused this particular workflow to take forever to run. However, i've removed the dependencies and it's back to the original build.gradle file because i realised that i don't need the new dependencies that were added (deleted the 2 lines and pushed it but it's still taking forever to run) .It's not failing, just taking forever to run... If I can't pass these runs, i won't be able to merge the PR.

**Comment by OrangeCatLoves** [comment]
I believe it shouldn't be taking even more than 20 minutes because usually it finishes in less than 2 minutes.

**Comment by damithc** [comment]
@OrangeCatLoves Probably something wrong with this specific run. Try stopping it and running it again?

**Comment by OrangeCatLoves** [comment]
@damithc Hi, i've re-run this about 4 times, and it's still taking forever to run.

**Comment by damithc** [comment]
@OrangeCatLoves Share the link to that PR so that others can see exactly what could be the culprit.

**Comment by OrangeCatLoves** [comment]
Here's the PR link. Thanks!
https://github.com/AY2425S1-CS2103T-F10-3/tp/pull/74

**Comment by damithc** [comment]
Since it is stuck in testing stage, perhaps try commenting out the tests you added/changed, just to see if that makes a difference?

**Comment by OrangeCatLoves** [comment]
The test finishes immediately because I commented out PersonCardTest.java which was the issue. However, the reason why I wrote this file was so that I can test the PersonCard.java - it is causing my codecov to fail at 33.33%. I've made several changes to the PersonCard.java and am attempting to write tests for that file to increase code coverage. At least the problem as to why the workflow took forever to run has surfaced. Thanks! I suspect it might be because I forgot to close the temporary application in the tests. Will look into it!
